diff --git a/llvm/lib/Target/SPIRV/SPIRVUtils.cpp b/llvm/lib/Target/SPIRV/SPIRVUtils.cpp
index 93ba63824561..ec439eb8e29d 100644
--- a/llvm/lib/Target/SPIRV/SPIRVUtils.cpp
+++ b/llvm/lib/Target/SPIRV/SPIRVUtils.cpp
@@ -199,6 +199,10 @@ std::string getStringImm(const MachineInstr &MI, unsigned StartIndex) {
 
 std::string getStringValueFromReg(Register Reg, MachineRegisterInfo &MRI) {
   MachineInstr *Def = getVRegDef(MRI, Reg);
+  if (Def && Def->getOpcode() == TargetOpcode::G_CONSTANT &&
+      Def->getOperand(1).getCImm()->isZero()) {
+    return {};
+  }
   assert(Def && Def->getOpcode() == TargetOpcode::G_GLOBAL_VALUE &&
          "Expected G_GLOBAL_VALUE");
   const GlobalValue *GV = Def->getOperand(1).getGlobal();
